### PR TITLE
Refactored to use a common <OrganizationPicker /> component

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description

This fixes a code climate issue reported about duplicate code.

## Related Tickets & Documents

#3739

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

I tested locally and UI renders the same for the org picker when writing a new post or creating a listing.

Before

image

image

After

image

image

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Happy cat with headphones on

### CODE REVIEW

**Comment by nickytonline** [comment]
WIP. I still have one other change to make.

**Comment by rhymes** [comment]
@nickytonline I added the WIP tag :)

**Comment by nickytonline** [comment]
The two new duplicate issues in Code climate can be ignored, as it's in regards to prop types. I was going to do more refactoring in this PR, but it means refactoring articleForm.jsx and listingForm.jsx which I think each merit their own PR. Once those are refactored, the prop types dupes error in code climate should disappear.

File: app/javascript/article-form/articleForm.jsx [hunk 1]
```diff
@@ -13,22 +13,23 @@ import Title from './elements/title';
...
 import ImageManagement from './elements/imageManagement';
 import MoreConfig from './elements/moreConfig';
-import OrgSettings from './elements/orgSettings';
 import Errors from './elements/errors';
 import KeyboardShortcutsHandler from './elements/keyboardShortcutsHandler';
 import Tags from '../shared/components/tags';
+import { OrganizationPicker } from '../organization/OrganizationPicker';
 
-const setupImageButton = ({ className = '', imgSrc, imgAltText = '', onClickCallback }) => {
+const setupImageButton = ({
```
Comment (nickytonline) [inline app/javascript/article-form/articleForm.jsx]: Just some eslint/prettier action happening here.

File: app/javascript/article-form/articleForm.jsx [hunk 2]
```diff
@@ -360,14 +361,17 @@ export default class ArticleForm extends Component {
...
       organizations && organizations.length > 0 ? (
-        <OrgSettings
-          organizations={organizations}
-          organizationId={organizationId}
-          onToggle={this.handleOrgIdChange}
-        />
-      ) : (
-        ''
-      );
+        <div className="articleform__orgsettings">
```
Comment (nickytonline) [inline app/javascript/article-form/articleForm.jsx]: The actual refactor.

File: app/javascript/src/components/common-prop-types/organization-prop-type.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import PropTypes from 'prop-types';
+
+export const organizationPropType = PropTypes.shape({
```
Comment (nickytonline) [inline app/javascript/src/components/common-prop-types/organization-prop-type.js]: Made an organization prop type that can be used throughout the project.

**Comment by nickytonline** [comment]
I just need to update this PR with the merged changes from #4217 and then this is good to go.

**Comment by nickytonline** [comment]
Updated and good to go.

**Comment by nickytonline** [comment]
I see that #4217 caused a an issue in the editor, so I’ll hold off on this PR until that’s all sorted and then I’ll rebase again to get this in.

**Comment by nickytonline** [comment]
Looks like #4505 fix the article clone issue. I've rebased and this should be good to go. The functionality is the same. The only difference is there is an `<OrganizationPicker />` component now that is reused.

**Comment by nickytonline** [comment]
Hmm, when publishing an article when an organization is selected, it gets published just for the user, not the org. Gonna see what I broke ;)

File: app/javascript/src/components/common-prop-types/organization-prop-type.js (hunk 3 above)
Comment (benhalpern) [reply app/javascript/src/components/common-prop-types/organization-prop-type.js]: Nice

**Pull request merged**